dot/parachain/collator-protocol/register.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package collatorprotocol
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/libp2p/go-libp2p/core/protocol"
+)
+
+func Register(net Network, protocolID protocol.ID, overseerChan chan<- any) (*CollatorProtocolValidatorSide, error) {
+
+	cpvs := CollatorProtocolValidatorSide{
+		SubSystemToOverseer: overseerChan,
+	}
+
+	// register collation protocol
+	err := net.RegisterNotificationsProtocol(
+		protocolID,
+		network.CollationMsgType,
+		getCollatorHandshake,
+		decodeCollatorHandshake,
+		validateCollatorHandshake,
+		decodeCollationMessage,
+		cpvs.handleCollationMessage,
+		nil,
+		MaxCollationMessageSize,
+	)
+	if err != nil {
+		// try with legacy protocol id
+		err1 := net.RegisterNotificationsProtocol(
+			protocol.ID(LEGACY_COLLATION_PROTOCOL_V1),
+			network.CollationMsgType,
+			getCollatorHandshake,
+			decodeCollatorHandshake,
+			validateCollatorHandshake,
+			decodeCollationMessage,
+			cpvs.handleCollationMessage,
+			nil,
+			MaxCollationMessageSize,
```

Comment:
is it not possible to register both protocols instead of one or another?

Reply:
It is the same protocol. The protocol ID has changed so we are trying both protocol IDs, for node that might be using old protocol IDs still.

Reply:
but you only try the legacy if the previous registration failed. I was thinking about listening to both protocols id